python/ray/data/datasource/parquet_datasource.py

```diff
@@ -374,13 +374,14 @@ def _read_pieces(
 
     logger.debug(f"Reading {len(pieces)} parquet pieces")
     use_threads = reader_args.pop("use_threads", False)
+    batch_size = reader_args.pop("batch_size", PARQUET_READER_ROW_BATCH_SIZE)
```

Comment:
The kwargs passing through the read_parquet() are for https://arrow.apache.org/docs/python/generated/pyarrow.parquet.read_table.html, which doesn't have a "batch_size".

Reply:
Our documentation needs to be fixed. It's actually https://arrow.apache.org/docs/python/generated/pyarrow.dataset.Scanner.html#pyarrow.dataset.Scanner.from_fragment, but not https://arrow.apache.org/docs/python/generated/pyarrow.parquet.read_table.html. See https://github.com/ray-project/ray/issues/30915#issuecomment-1354299322 for details.

Reply:
The question is if it would make sense it to pass args from `read_parquet()` to things other than `read_table()` API.

Reply:
I think it makes sense. We are actually passing kwargs to [pyarrow.parquet.ParquetDataset](https://github.com/ray-project/ray/blob/master/python/ray/data/datasource/parquet_datasource.py#L197-L201) and [Scanner.from_fragment](https://arrow.apache.org/docs/python/generated/pyarrow.dataset.Scanner.html#pyarrow.dataset.Scanner.from_fragment). These APIs do not have same arguments as `read_table()`.

Reply:
@jianoaix The `ray.data.read_parquet()` API is more of a distributed analog for `pyarrow.parquet.ParquetDataset`, where we expose certain features that the underlying `ParquetDataset` provides (e.g. reading path-based partition columns into the data, supporting zero-read filter pushdown on partition columns, etc.). We do actually have a distributed analog for `pyarrow.parquet.read_table()`, and that's `ray.data.read_parquet_bulk()`, which doesn't use `pyarrow.parquet.ParquetDataset` and instead directly uses `pyarrow.parquet.read_table()`.

For this API, I think that directing users to `pyarrow.dataset.Scanner.from_fragment()` for `**arrow_parquet_args` and to `pyarrow.parquet.ParquetDataset` for `dataset_kwargs` would be best. Going forward, if we continue to build out our own format-agnostic partitioning machinery, we should eventually consider switching to `pyarrow.parquet.read_table()` if/when we achieve feature parity.

Reply:
Good to know that. IIUC, the difference between read_table v.s. ParuqetDataset/Scanner is whether we stream read a single file. So in `read_parquet_bulk()`, when reading a single file, it's not streamed, whereas in `read_parquet()`, it is streaming. It looks to me converging on ParuqetDataset/Scanner for streaming single file is a good option for all those read APIs.

For this PR itself, since it's leveraging existing arg passing, LG to move forward. We may discuss in followup in how to make the APIs / arg passing better.

Reply:
The key difference is actually whether or not we're using Parquet's dataset stuff, which gives us a bunch of partitioning machinery. The streaming vs. full read was just an implementation detail for getting the performance out of `read_parquet_bulk()` that Amazon needed, since this was before we fixed the buffering for the streaming Parquet read; we could probably move the `read_parquet_bulk()` to a streaming read with a buffer size set and get the same performance as the full read.